phase/prepare_arm.go

```diff
@@ -0,0 +1,55 @@
+package phase
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/k0sproject/k0sctl/config"
+	"github.com/k0sproject/k0sctl/config/cluster"
+)
+
+// PrepareArm implements a phase which fixes arm quirks
+type PrepareArm struct {
+	GenericPhase
+
+	hosts cluster.Hosts
+}
+
+// Title for the phase
+func (p *PrepareArm) Title() string {
+	return "Prepare ARM nodes"
+}
+
+// Prepare the phase
+func (p *PrepareArm) Prepare(config *config.Cluster) error {
+	p.Config = config
+
+	p.hosts = p.Config.Spec.Hosts.Filter(func(h *cluster.Host) bool {
+		arch := h.Metadata.Arch
+		return h.Role != "worker" && (arch == "arm" || arch == "arm64")
```

Comment:
Note to self: I think this needs `aarch64` as well.

Reply:
Possibly `armv7l` and some others... maybe just `strings.HasPrefix` with `arm` and `aarch`, then figure out 32/64bit separately to choose if `etcd_unsupported_arch` should be set to `arm` or `arm64`.

https://etcd.io/docs/v3.5/op-guide/supported-platform/ here's the list of supported platforms and which ones need the env.

I think it needs to print a warning too.